src/test/java/core/basesyntax/FruitShopServiceTest.java

```diff
@@ -0,0 +1,115 @@
+    private FruitShopService fruitShopService;
+    private Map<FruitTransaction.Operation, DataHandler> enumHandlerMap;
+
+    @BeforeAll
+    static void createStorage() {
+        Storage.createMap();
+    }
+
+    @BeforeEach
+    void setUp() {
+        enumHandlerMap = initializeEnumHandlerMap();
+        DataHandlerStrategy dataHandlerStrategy = new TestDataHandlerStrategy(enumHandlerMap);
+        transactions = new ArrayList<>();
+        fruitShopService = new FruitShopServiceImpl(dataHandlerStrategy);
+    }
+
+    private Map<FruitTransaction.Operation, DataHandler> initializeEnumHandlerMap() {
+        Map<FruitTransaction.Operation, DataHandler> enumHandlerMap = new HashMap<>();
+        enumHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceDataHandler());
+        enumHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyDataHandler());
+        enumHandlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseDataHandler());
+        enumHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnDataHandler());
+        return enumHandlerMap;
+    }
+
+    @Test
+    void updateData_validTransactions_okay() {
+        transactions.add(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, APPLE, APPLE_QUANTITY));
+        transactions.add(
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, APPLE, APPLE_QUANTITY));
+        transactions.add(
+                new FruitTransaction(FruitTransaction.Operation.RETURN, APPLE, APPLE_QUANTITY));
+        transactions.add(
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, APPLE, APPLE_QUANTITY));
+        transactions.add(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, BANANA, BANANA_QUANTITY));
+        fruitShopService = new FruitShopServiceImpl(new TestDataHandlerStrategy(enumHandlerMap));
+        fruitShopService.updateData(transactions);
+        assertEquals(EXPECTED_APPLE_QUANTITY, Storage.getStorage().get(APPLE));
+        assertEquals(EXPECTED_BANANA_QUANTITY, Storage.getStorage().get(BANANA));
+    }
+
+    @Test
+    void updateData_negativeQuantity_okay() {
+        transactions.add(new FruitTransaction(
+                FruitTransaction.Operation.BALANCE, APPLE, WRONG_QUANTITY));
+        fruitShopService = new FruitShopServiceImpl(new TestDataHandlerStrategy(enumHandlerMap));
+        assertThrows(FruitsQuantityException.class,
+                () -> fruitShopService.updateData(transactions));
+    }
+
+    @Test
+    void updateData_nullTransactions_okay() {
+        transactions = null;
+        assertThrows(TransactionException.class,
+                () -> fruitShopService.updateData(transactions));
+        assertEquals(EMPTY_STORAGE_SIZE, Storage.getStorage().size());
+    }
+
+    @AfterEach
+    void cleanStorage() {
+        Storage.clear();
+    }
+
+    private static class TestDataHandlerStrategy implements DataHandlerStrategy {
```

Comment:
Do you need it? Why this is here?

Reply:
I use it for independent testing) Am I suppose to use Strategy directly from Fruit Shop?

Reply:
That's enough add one operation to test that logic related to the FruitShop is working fine 
All this private class and method for initialize all operation is redundant and make your code and test harder for improvements/changes

Reply:
Not resolved